cpp/src/io/orc/reader.cu

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (c) 2024, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "io/orc/reader_impl.hpp"
+#include "io/orc/reader_impl_helpers.hpp"
+
+namespace cudf::io::orc::detail {
+
+// Constructor and destructor are defined within this translation unit.
+reader::reader()  = default;
+reader::~reader() = default;
```

Comment:
these are usually defined in the header

Reply:
This is to support forward declaration so you can declare the reader class without definition.

Reply:
I see, because of impl :+1: